### ISSUE TITLE
Not flagging several miss-spellings of "corresponding"

### ISSUE THREAD

**Issue by andy-z** [streetsidesoftware/vscode-spell-checker#802]
Out of "coresponding, correspending, corrusponding, correspendi" only the last one is flagged. This is with C++ mode:

![image](https://user-images.githubusercontent.com/9602964/114062357-46eff700-984c-11eb-8293-296d20188b32.png)

**Comment by Jason3S** [comment]
@andy-z,

This is caused by the `allowCompoundWords` setting that is on by default for C++.

To turn it off:

Section in a `cspell.json`
```
{
    "languageSettings": [
        {
            "languageId": "c,cpp",
            "allowCompoundWords": false,
        }
    ]
}
```

VS Code Settings:
```
    "cSpell.languageSettings": [
        {
            "languageId": "c,cpp",
            "allowCompoundWords": false,
        }
    ]
}
```

**Comment by andy-z** [comment]
Thanks for suggestions, I tried to add it to settings.json but it does not change a thing. Besides I suspect turning it off will flag a lot of variable name, will it not?

**Comment by Jason3S** [comment]
> Thanks for suggestions, I tried to add it to settings.json but it does not change a thing. 

There is a bug in VS Code, that if you add a new section to `settings.json`, it won't get picked up until VS Code is restarted.

> Besides I suspect turning it off will flag a lot of variable name, will it not?

Yes, most likely.

**Comment by maroc81** [comment]
I just found this after trying for a couple of hours to figure out why "satisifed" was not flagged as a bad word.  Turns out this setting was the problem.  Turning it off revealed a number other words that were being missed.  Any reason it's on by default for C++? I'm guessing it can't be restricted to code only (i.e. don't allow compound words in comments)?

**Comment by Jason3S** [comment]
@maroc81,

Because it is a breaking change, I'll turn it off by default for all languages on the next major release.

**Comment by Jason3S** [comment]
It is now off by default.
